stutterhold/Source/PhaseVocodeur/PhaseVocodeur.cpp

```diff
@@ -59,32 +164,28 @@ PhaseVocodeur::~PhaseVocodeur()
     delete[] fft_out;
 }
 
-// ==============================================================================
+/*
+ ==============================================================================
+ Writing and reading.
+ ==============================================================================
+ */
 
 void PhaseVocodeur::push(float input_sample)
 {
-    /* write to buffers */
+    // Takes input sample and writes it into all the write buffers.
     auto ola_in_w = ola_in.getArrayOfWritePointers();
+    
     for (int b = 0; b < num_ola_frames; b++)
     {
         ola_in_w[b][rw[b]] = input_sample;
-        // IF rw[b] is at the end of a frame THEN process
+        // IF rw[b] is at the end of a frame THEN process.
         if (rw[b] == ola_size-1)
         {
             spectral_routine(b);
         }
     }
 }
 
-void PhaseVocodeur::advance()
-{
-    /* advances rw positions */
-    for (int k = 0; k < rw.size(); k++)
-    {
-        rw[k] = (rw[k] + 1) % ola_size;
-    }
-}
-

```

Comment:
From here on down it's mostly just rearranging. I didn't remove anything, just moved stuff.